modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/model/CreateDetectorMappingRequest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.model;
+
+import com.expedia.adaptivealerting.modelservice.web.RequestValidator;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Data
+public class CreateDetectorMappingRequest {
+    private Expression expression;
+    private Detector detector;
+    private User user;
+
+    @Autowired
+    private RequestValidator requestValidator;
```

Comment:
As a general rule, you can't autowire into non-components. (That's not entirely true--there's an AOP-based way to do this using `@Configurable`--but it doesn't appear you're using that here, and it requires a more advanced AOP configuration that I'd prefer to avoid unless necessary.) So I would expect `requestValidator` to be `null` after Spring deserializes it.

Instead consider simply making the request validator a utility class with a collection of static methods.

Reply:
Moved it to util.